portlist/portlist_windows.go

```diff
@@ -92,29 +95,112 @@ func (im *windowsImpl) AppendListeningPorts(base []Port) ([]Port, error) {
 		}
 		ret = append(ret, m.port)
 	}
+
 	return sortAndDedup(ret), nil
 }
 
-func procNameOfPid(pid int) string {
-	const da = windows.PROCESS_QUERY_LIMITED_INFORMATION
-	h, err := syscall.OpenProcess(da, false, uint32(pid))
+// pidInfo is a mapping of process IDs to process names that incorporates
+// information from Windows' Service Control Manager. We use this information
+// to disambiguate svchost processes (when possible).
+type pidInfo map[uint32]string
+
+// newPidInfo instantiates a pidInfo and pre-populates it with PID mappings for
+// all Windows services that are currently running.
+func newPidInfo() (result pidInfo) {
```

Comment:
can we only do all this Service Manager work if the process name is "svchost"? Seems overkill to do each time "python -m SimpleHTTPServer 8080" appears for a second and then disappears and then comes back. If we know right away that it's "python", then all the Service Manager querying is for nothing?

Reply:
Yeah, we can do that.